vendor/processors/authorizenet.py

```diff
@@ -421,31 +446,25 @@ def subscription_update_payment(self, receipt, subscription_id):
 
         self.check_subscription_response(response)
 
-        if not self.transaction_submitted:
-            return None
-
-        receipt.meta[f"{timezone.now():%Y-%m-%d %H:%M}"] = {'raw': str({**self.transaction_message, **response})}
+        receipt.meta[f"payment-update-{timezone.now():%Y-%m-%d %H:%M}"] = {'raw': str({**self.transaction_message, **response})}
         receipt.save()
 
-        payment = Payment.objects.get(success=True, transaction=receipt.transaction, invoice=receipt.order_item.invoice)
-
-        subscription_info = self.subscription_info(subscription_id)
+        subscription_info = self.subscription_info(receipt.transaction)
 
         account_number = subscription_info['subscription']['profile']['paymentProfile']['payment']['creditCard'].__dict__.get('cardNumber', None)
         if account_number:
-            payment.result['account_number'] = account_number.text
+            self.payment.result['account_number'] = account_number.text
             
-        account_type = payment.result['account_type'] = subscription_info['subscription']['profile']['paymentProfile']['payment']['creditCard'].__dict__.get('accountType', None)
+        account_type = subscription_info['subscription']['profile']['paymentProfile']['payment']['creditCard'].__dict__.get('accountType', None)
```

Comment:
good to know, I did not know that it was risky. Gonna read on it. I made the change:
```
        account_number = getattr(subscription_info['subscription']['profile']['paymentProfile']['payment']['creditCard'], 'cardNumber', None)
        if account_number:
            self.payment.result['account_number'] = account_number.text
            
        account_type = getattr(subscription_info['subscription']['profile']['paymentProfile']['payment']['creditCard'], 'accountType', None)
        if account_type:
            self.payment.result['account_type'] = account_type.text
```

Reply:
Yeah, anything that leads with an '_' is considered private and should be left alone (not overridden) since it's used internally and anything that uses double underbars on either side is a "built-in" that it typically viewed as off-limits.